src/main/java/org/folio/search/model/rest/response/SearchResult.java

```diff
@@ -1,11 +1,14 @@
 package org.folio.search.model.rest.response;
 
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
 import lombok.Data;
-import lombok.EqualsAndHashCode;
-import lombok.NoArgsConstructor;
 
 @Data
-@NoArgsConstructor
-@EqualsAndHashCode(callSuper = true)
-public class SearchResult extends PagedResult<Object> {
+@AllArgsConstructor(staticName = "of")
+public class SearchResult implements PagedResult<Map<String, Object>> {
+
+  private final long totalRecords;
+  private final List<Map<String, Object>> instances;
```

Comment:
Consider renaming to `records` if the same generic idea applies.

Reply:
In current implementation - I am trying to support the contract, that is used in existing inventory search and it returns only instances
Also, during working on swagger documentation - map has been replaced with instance object